libfuse/folderlist.go

```diff
@@ -67,21 +91,48 @@ func (fl *FolderList) Lookup(ctx context.Context, req *fuse.LookupRequest, resp
 		return specialNode, nil
 	}
 
-	if child, ok := fl.folders[req.Name]; ok {
-		return child, nil
-	}
-
 	// Shortcut for dreaded extraneous OSX finder lookups
 	if strings.HasPrefix(req.Name, "._") {
 		return nil, fuse.ENOENT
 	}
 
 	h, err := libkbfs.ParseTlfHandle(
 		ctx, fl.fs.config.KBPKI(), req.Name, fl.public)
+
+	addToFavorite := func() (err error) {
```

Comment:
I think this would be better as a real function on `FolderList` because a) it would keep `Lookup` smaller and more readable, and b) it's only used in the `nil` case below, so we wouldn't waste time constructing the closure for all the other cases.